### PR TITLE
Update graphics.lua

### PR DESCRIPTION
**Pull request by alxthmpson**
I added comments for better readability and changed the love.graphics.setColorHSV function to normalize RGB values between 0 and 1. I also used consistent indentation for a cleaner look

### CODE REVIEW

**Comment by alxthmpson** [comment]
Hi @bhaskar10h, I have some questions about - [`FrameTestScene.lua`](https://github.com/bhaskar10h/dlg/blob/master/src/game/scenes/FrameTestScene.lua)

- How is the size of the frame restricted to non-negative values during resizing in the `FrameTestScene` class?
- How does the clipped property of the frame affect its rendering in the `FrameTestScene` class, and what visual difference does it make when toggling this property?

Thanks!

File: src/engine/graphics.lua [hunk 1]
```diff
@@ -46,5 +46,5 @@ end
 -- Set color with HSV values
 function love.graphics.setColorHSV(h, s, v, a)
     local r, g, b = std.hsvToRgb(h, s, v)
-    love.graphics.setColor(r, g, b, a or 255)
+    love.graphics.setColor(r / 255, g / 255, b / 255, a and a / 255 or 1)
```
Comment (alxthmpson) [inline src/engine/graphics.lua:49]: @bhaskar10h, Let me know if you want me to include a way to work with HSL here.

Comment (alxthmpson) [reply src/engine/graphics.lua:49]: @bhaskar10h, I may also include CMYK if you want me to.

**Comment by bhaskar10h** [comment]
> Hi @bhaskar10h, I have some questions about - [`FrameTestScene.lua`](https://github.com/bhaskar10h/dlg/blob/master/src/game/scenes/FrameTestScene.lua)
> 
> * How is the size of the frame restricted to non-negative values during resizing in the `FrameTestScene` class?
> * How does the clipped property of the frame affect its rendering in the `FrameTestScene` class, and what visual difference does it make when toggling this property?
> 
> Thanks!

The code checks if either the x or y component of the frame size becomes negative after resizing. If so, it sets the respective component to 0, effectively preventing the frame from having negative dimensions.

When clipped is set to true, any content drawn outside the bounds of the frame will be clipped, meaning it will not be rendered

When toggling the clipped property from true to false, you may not notice a visual difference. However, if the content were to extend beyond the bounds of the frame, toggling clipped to false would cause the extended content to be visible outside the frame.

File: src/engine/graphics.lua (hunk 1 above)
Comment (bhaskar10h) [reply src/engine/graphics.lua:49]: > @bhaskar10h, Let me know if you want me to include a way to work with HSL here.

CMYK is too much for this project and I don't want  this to be bright vibrant.

**Comment by alxthmpson** [comment]
@bhaskar10h, Did you use ChatGPT for your response?

**Comment by bhaskar10h** [comment]
Hey Alex,

I am not a good English user, so i had to put some explanations in my own
words and paraphrase it  using chatGPT

**Comment by alxthmpson** [comment]
@bhaskar10h, Appreciate the honesty!

**Comment by bhaskar10h** [comment]
Thank you.

**Pull request merged**